### PR TITLE
Preferences

### PR DESCRIPTION
**Pull request by jaadams5**
Lint, typos and cleanup so root\settings remains empty after tests.  Cleanup should happen automatically but tests that create multiple preferences files may have to do additional work.  A human who wants to examine a preference file after a test is probably not going to be happy.  Draft status as I ponder these things.

### CODE REVIEW

**Comment by jaadams5** [comment]
root\settings remains empty after tests

added tests for more coverage

commented about ways to make tests leak, but these don't seem to.

when tests pass on github, will change draft status

**Comment by jaadams5** [comment]
and nothing has been done about the security scanner item that was why I went down this rabbit hole.  Next one.

**Comment by jaadams5** [comment]
Ans maybe I have made the scanner happier so one more commit

**Comment by jaadams5** [comment]
Probably easier ways to resolve conflicts but...

File: test/net/sourceforge/kolmafia/extensions/ClearSharedState.java [hunk 1]
```diff
@@ -29,4 +34,24 @@ public void afterAll(ExtensionContext context) {
     KoLmafia.lastMessage = "";
     KoLmafia.forceContinue();
   }
+
+  public static void deleteEm(String user) {
```
Comment (gausie) [inline test/net/sourceforge/kolmafia/extensions/ClearSharedState.java:38]: Why is this called `deleteEm`?

Comment (gausie) [reply test/net/sourceforge/kolmafia/extensions/ClearSharedState.java]: Ohhh that was silly of me

**Review by gausie — APPROVED** [review]
Approved. What should live inside `extensions`? If this is purely test related, shouldn't it live in the test package?

**Comment by jaadams5** [comment]
I was following he3s lead and just put code that dealt with preferences leaking in tests in the place where there was already code to do that.  Worthy discussion topic.

**Pull request merged**